app/models/user.rb

```diff
@@ -173,6 +175,14 @@ class User < ApplicationRecord
   # minimum number of submitted stories before checking self promotion
   MIN_STORIES_CHECK_SELF_PROMOTION = 2
 
+  def underscores_and_dashes_in_username
+    username = self.username.split(/_|-/)
+    return if username.length < 2
+
+    collisions = User.where('username REGEXP ?', username.join("[-_]"))
+    errors.add(:username, "is too similar to other usernames") if collisions.any?
```

Comment:
```suggestion
    errors.add(:username, "is already in use (perhaps swapping _ and -)") if collisions.any?
```

Trying to be explicit about what "similarity" means.

Reply:
In fact, I was hoping for this suggestion, so I could have an example message from you. 😅